ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+                           sizeof(bytes_per_camp));
+
+                bytes_acumulats += bytes_per_camp;
+                pMMBDXP->nFields++;
+            } while (bytes_acumulats < pMMBDXP->BytesPerRecord);
+        }
+    }
+
+    if (pMMBDXP->nFields != 0)
+    {
+        pMMBDXP->pField = MM_CreateAllFields(pMMBDXP->nFields);
+        if (!pMMBDXP->pField)
+        {
+            fclose_function(pf);
+            return 1;
+        }
+    }
+    else
+        pMMBDXP->pField = nullptr;
+
+    fseek_function(pf, 32, SEEK_SET);
+    for (nIField = 0; nIField < pMMBDXP->nFields; nIField++)
+    {
+        if (1 != fread_function(pMMBDXP->pField[nIField].FieldName,
+                                MM_MAX_LON_CLASSICAL_FIELD_NAME_DBF, 1, pf) ||
+            1 != fread_function(&(pMMBDXP->pField[nIField].FieldType), 1, 1,
+                                pf) ||
+            1 != fread_function(&(pMMBDXP->pField[nIField].reserved_1), 4, 1,
+                                pf) ||
+            1 != fread_function(&(pMMBDXP->pField[nIField].BytesPerField), 1, 1,
+                                pf) ||
+            1 != fread_function(&(pMMBDXP->pField[nIField].DecimalsIfFloat), 1,
+                                1, pf) ||
+            1 != fread_function(&(pMMBDXP->pField[nIField].reserved_2), 13, 1,
+                                pf) ||
+            1 != fread_function(&(pMMBDXP->pField[nIField].MDX_field_flag), 1,
+                                1, pf))
+        {
+            free(pMMBDXP->pField);
+            fclose_function(pf);
+            return 1;
+        }
+
+        if (pMMBDXP->pField[nIField].FieldType == 'F')
+            pMMBDXP->pField[nIField].FieldType = 'N';
+
+        pMMBDXP->pField[nIField]
+            .FieldName[MM_MAX_LON_CLASSICAL_FIELD_NAME_DBF - 1] = '\0';
+        if (EQUAL(pMMBDXP->pField[nIField].FieldName,
+                  szMMNomCampIdGraficDefecte))
+            pMMBDXP->IdGraficField = nIField;
+
+        if (pMMBDXP->pField[nIField].BytesPerField == 0)
+        {
+            if (!MM_ES_DBF_ESTESA(pMMBDXP->dbf_version))
+            {
+                free(pMMBDXP->pField);
+                fclose_function(pf);
+                return 1;
+            }
+            if (pMMBDXP->pField[nIField].FieldType != 'C')
+            {
+                free(pMMBDXP->pField);
+                fclose_function(pf);
+                return 1;
+            }
+
+            memcpy(&pMMBDXP->pField[nIField].BytesPerField,
+                   (char *)(&pMMBDXP->pField[nIField].reserved_2) + 3,
+                   sizeof(MM_BYTES_PER_FIELD_TYPE_DBF));
+        }
+
+        if (nIField)
+            pMMBDXP->pField[nIField].AcumulatedBytes =
+                (pMMBDXP->pField[nIField - 1].AcumulatedBytes +
+                 pMMBDXP->pField[nIField - 1].BytesPerField);
+        else
+            pMMBDXP->pField[nIField].AcumulatedBytes = 1;
+
+        if (pszRelFile)
+        {
+            sprintf(section, "TAULA_PRINCIPAL:%s",
```

Comment:
```suggestion
            snprintf(section, sizeof(section), "TAULA_PRINCIPAL:%s",
```

Reply:
I converted all sprintf to snprintf. Thanks.